### PR TITLE
CLI and test improvements

### PR DESCRIPTION
**Pull request by dkcumming**
### This PR
- changes default output for `parse` and `run` to `pretty`  so that the flag can be omitted
- removes flag `--spec-file` for positional argument `spec_file`
- Changes the call for `make test-integration` as 1 call to `pytest` for efficiency (details below)

#### Changes to `pytest`
[Previously](https://github.com/runtimeverification/mir-semantics/pull/270/files#diff-7d1398ca5694151a98067ce31bf94559eb7e91340972dd278e224f50551ab83dL38) `make test` would call `pytest` 3 times, this would result in the definitions being compiled 3 times ([link1](https://github.com/runtimeverification/mir-semantics/blob/310510c24b6b62c19b63ab3662d1e8919bddd262/kmir/src/tests/integration/test_parse.py#L23), [link2](https://github.com/runtimeverification/mir-semantics/blob/310510c24b6b62c19b63ab3662d1e8919bddd262/kmir/src/tests/conftest.py#L69-L81)). However the definitions can be reused across the same session, so a single call to `pytest` that bundles the 3 tests is more efficient. Here is the difference in time:
```
OLD WAY (3 calls to `pytest`):
real    2m51.561s
user    26m15.991s
sys     1m37.795s

NEW WAY (1 call to `pytest` bundling the tests):
real    1m55.352s
user    12m5.252s
sys     0m46.990s
```
A downside is that this means that the [output of the tests is not separated](https://github.com/runtimeverification/mir-semantics/actions/runs/6977808091/job/18988193184?pr=270#step:5:11458). I think this is a fine trade off as we can run `make test-integration-parse`, `make test-integration-run`, `make test-integration-prove` if we wish to isolate the output.

### CODE REVIEW

**Review by yanliu18 — CHANGES_REQUESTED** [review]

File: kmir/Makefile [hunk 1]
```diff
@@ -35,7 +35,12 @@ test: test-unit test-integration
 test-unit: poetry-install
 	$(POETRY_RUN) pytest src/tests/unit --maxfail=1 --verbose $(TEST_ARGS)
 
-test-integration: test-integration-parse test-integration-run test-integration-prove
+test-integration: poetry-install
+	$(POETRY_RUN) pytest                                                  \
+        src/tests/integration/test_parse.py                               \
+        src/tests/integration/test_run.py                                 \
+        src/tests/integration/test_prove.py                               \
+        --numprocesses=4 --durations=0 --maxfail=1 --verbose $(TEST_ARGS) \
```
[the remark this answers is not in the document]
Comment (dkcumming) [reply kmir/Makefile:43]: From going through the [documentation here](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html), this seemed to be the way to use the fixtures. I did try to see if there was a way to manually split the reports for the provided tests also but I didn't see a way of doing that. @tothtamas28 do you have any recommendation for how to achieve both?

[the remark this answers is not in the document]
Comment (tothtamas28) [reply kmir/Makefile:43]: > do you have any recommendation for how to achieve both?

The options I see are:
1. Do not build the definition as part of test setup, but before that as a manual step. Then run the tests separately.
2. Run all tests in a single run, post-process the test report.

If you choose (2), consider outputting the test report in some structured format: https://docs.pytest.org/en/latest/how-to/output.html

**Comment by yanliu18** [comment]
@dkcumming thanks. The short summary looks good. Would it be possible to add a bit more information?
For example, line 11500 in the [test report](https://github.com/runtimeverification/mir-semantics/actions/runs/7083153222/job/19275076824?pr=270#step:5:11500) is 
```SKIPPED [767] src/tests/integration/test_parse.py:55: Skipped```

can it be the following:
```SKIPPED [767/2550] src/tests/integration/test_parse.py::test_compiletest```

**Comment by dkcumming** [comment]
@yanliu18 I think added the passed tests is definitely not what we want, as I mentioned above it will list them all individually, so that is hundreds maybe thousands of lines of 
```
PASSED src/tests/integration/test_name
```

**Comment by yanliu18** [comment]
@dkcumming  Sorry, a mistake. I meant `-ra`.

**Comment by dkcumming** [comment]
@yanliu18 Okay I thought you meant `-rsp` again, I think I saw `-rs_` and didn't register the `a` at the end. If you agree that `-ra` is best can you approve?

**Comment by yanliu18** [comment]
OK, I can approve this change.
There should be a better way of setting up the tests when the [refactor PR](https://github.com/runtimeverification/mir-semantics/pull/283) is done.
Can you update the branch?

**Review by yanliu18 — APPROVED** [review]

**Pull request merged**